libappstream-builder/plugins/asb-plugin-sugar-activity.c

```diff
@@ -0,0 +1,286 @@
+ */
+void
+asb_plugin_add_globs (AsbPlugin *plugin, GPtrArray *globs)
+{
+	// the _SUGAR_ACTIVITY_GLOB is a subset of the icon glob
+	asb_plugin_add_glob (globs, _SUGAR_ACTIVITY_ICON_GLOB);
+	asb_plugin_add_glob (globs, _SUGAR_ACTIVITY_LINFO_GLOB);
+}
+
+/**
+ * _asb_plugin_is_actinfo
+ */
+static gboolean
+_asb_plugin_is_actinfo (const gchar *filename)
+{
+	if (asb_plugin_match_glob (_SUGAR_ACTIVITY_GLOB, filename))
+		return TRUE;
+	return FALSE;
+}
+
+/**
+ * _asb_plugin_is_linfo
+ */
+static gboolean
+_asb_plugin_is_linfo (const gchar *filename)
+{
+	if (asb_plugin_match_glob (_SUGAR_ACTIVITY_LINFO_GLOB, filename))
+		return TRUE;
+	return FALSE;
+}
+
+/**
+ * asb_plugin_check_filename:
+ */
+gboolean
+asb_plugin_check_filename (AsbPlugin *plugin, const gchar *filename)
+{
+	return _asb_plugin_is_linfo (filename) ||
+	       _asb_plugin_is_actinfo (filename);
+}
+
+static gboolean
+_asb_plugin_load_sugar_icon (const gchar *path,
+			     AsIcon *ic) {
+	gsize len;
+	gchar *data = NULL;
+	GBytes *bytes = NULL;
+
+	if (!g_file_get_contents (path, &data, &len, NULL))
+		return FALSE;
+	printf ("ICON (got contents) = %s\n", path);
+
+	as_icon_set_kind (ic, AS_ICON_KIND_EMBEDDED);
+	bytes = g_bytes_new_take (data, len);
+	as_icon_set_data (ic, bytes);
+	as_icon_set_width (ic, 75);
+	as_icon_set_height (ic, 75);
+	return TRUE;
+}
+
+/**
+ * asb_plugin_process_actinfo:
+ */
+static gboolean
+asb_plugin_process_actinfo (AsbPlugin *plugin,
+			    AsbPackage *pkg,
+			    const gchar *filename,
+			    AsApp *app,
+			    GError **error)
+{
+	gsize len;
+	const gchar *tmp;
+	g_autofree gchar *data = NULL;
+	g_autofree gchar *id = NULL;
+	g_autoptr(GKeyFile) actinfo = NULL;
+
+	if (!g_file_get_contents (filename, &data, &len, error))
+		return FALSE;
+
+	actinfo = g_key_file_new ();
+	if (!g_key_file_load_from_data (actinfo, data, len,
+					G_KEY_FILE_NONE, error))
+		return FALSE;
+
+	if (!g_key_file_has_group (actinfo, "Activity"))
+		return FALSE;
+	if (!g_key_file_has_key (actinfo, "Activity", "bundle_id", error))
+		return FALSE;
+
+	as_app_add_category (AS_APP (app), "Sugar");
+	as_app_add_category (AS_APP (app), "SugarActivity");
+	as_app_set_kind (app, AS_APP_KIND_DESKTOP);
+
+	tmp = g_key_file_get_string (actinfo, "Activity", "bundle_id", NULL);
+	id = g_strdup_printf ("%s.activity.desktop", tmp);
+	as_app_set_id (app, id);
+
+	tmp = g_key_file_get_string (actinfo, "Activity", "name", NULL);
+	if (tmp != NULL)
+		as_app_set_name (app, NULL, tmp);
+
+	tmp = g_key_file_get_string (actinfo, "Activity", "summary", NULL);
+	if (tmp != NULL) {
+		as_app_set_comment (app, NULL, tmp);
+		g_autofree gchar *desc = NULL;
+		desc = as_markup_import (tmp,
+					 AS_MARKUP_CONVERT_FORMAT_SIMPLE,
+					 error);
+		if (desc != NULL)
+			as_app_set_description (app, NULL, desc);
+	}
+
+	tmp = g_key_file_get_string (actinfo, "Activity", "license", NULL);
+	if (tmp != NULL) {
+		as_app_set_project_license (app, tmp);
+		as_app_set_metadata_license (app, tmp);
```

Comment:
This isn't true; the metadata license is what the metadata is available under, and has to be a content license like CC0 rather than a code license like GPL-3.0+